### PR TITLE
Add support for STM32H733xx

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Summary

Support binaries for STMH733xx microcontrollers, and add a project for the H7 dev board.

### Changelist 

- Support ARM Cortex-M7
- Support STM33H7xx microcontrollers with SEGGER SystemView, FreeRTOS, and H7 HAL drivers
- Add project for H7 dev board

Note: CAN is not supported on H7 yet, since it ST's FD CAN HAL driver rather than ST's CAN HAL driver (which we use on the rest of our microcontrollers).

### Testing Done

- None

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -71,6 +71,20 @@
       "showDevDebugOutput": "none",
       "preLaunchTask": "Build: DIM.elf"
     },
+    {
+      "name": "Debug Embedded: h7dev",
```
Comment (will-chaba) [inline .vscode/launch.json:75]: Good for now, may need to re-think how to clearly list which boards are h7 vs F4 in the future

File: firmware/dev/h7dev/src/app/configs/App_SharedSignalConfig.h [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#pragma once
+
+struct H7DevWorld;
+#define World H7DevWorld
```
Comment (will-chaba) [inline firmware/dev/h7dev/src/app/configs/App_SharedSignalConfig.h:4]: should this have the exact same contents as firmware/dev/h7dev/src/app/configs/App_SharedStateMachineConfig.h?

**Review by will-chaba — APPROVED** [review]
Looks good, small concerns notes

File: firmware/CMakeLists.txt [hunk 3]
```diff
@@ -10,7 +10,14 @@ if("${TARGET}" STREQUAL "deploy")
     CPMAddPackage(
         NAME STM32CUBEF4
         GITHUB_REPOSITORY UBCFormulaElectric/STM32CubeF4
-        GIT_TAG 3e907f8
+        GIT_TAG master
```
[the remark this answers is not in the document]
Comment (gtaharaedmonds) [reply firmware/CMakeLists.txt:13]: yep good call

File: firmware/dev/h7dev/src/app/configs/App_SharedSignalConfig.h (hunk 2 above)
Comment (gtaharaedmonds) [reply firmware/dev/h7dev/src/app/configs/App_SharedSignalConfig.h]: just removed the shared code that needed these via cmake instead

**Pull request merged**